lib/ntlmauth/ntlmauth.h

```diff
@@ -32,21 +32,21 @@ extern "C" {
 #define NTLM_REQUEST_NON_NT_SESSION_KEY     0x400000
 
 /* NTLM error codes */
-#define NTLM_ERR_INTERNAL         -3
-#define NTLM_ERR_BLOB             -2
-#define NTLM_ERR_BAD_PROTOCOL     -1
-#define NTLM_ERR_NONE              0    /* aka. SMBLM_ERR_NONE */
-/* codes used by smb_lm helper */
-#define NTLM_ERR_SERVER            1    /* aka. SMBLM_ERR_SERVER   */
-#define NTLM_ERR_PROTOCOL          2    /* aka. SMBLM_ERR_PROTOCOL */
-#define NTLM_ERR_LOGON             3    /* aka. SMBLM_ERR_LOGON    */
-#define NTLM_ERR_UNTRUSTED_DOMAIN  4
-#define NTLM_ERR_NOT_CONNECTED     10
-/* codes used by mswin_ntlmsspi helper */
-#define NTLM_SSPI_ERROR         1
-#define NTLM_BAD_NTGROUP        2
-#define NTLM_BAD_REQUEST        3
-/* TODO: reduce the above codes down to one set non-overlapping. */
+enum class NtlmError
```

Comment:
This C++ code should not be surrounded by `extern "C" {}` but is.

I suspect this is not the only place where the updated header declarations have become C++ specific. Perhaps that wrapper should now be removed (or moved to become attached to a few function declarations that still need it)? To avoid a misunderstanding, I do not know the answer to that question. I only know that the current code looks self-contradictory to me.

Reply:
That is a very good question. In Squid's codebase there are 53 files containing extern "C" definitions. Many of them are in compat/ , include/ and lib/ , but 12 are in src/ .

I'll make a note to look into whether these can be removed in a following PR

Reply:
As for this specific change, I'm removing the extern linkage specifier. Thanks for catching this